File: Artist/app/src/main/java/com/example/artist/ui/topalbumslisting/TopAlbumsView.java
package com.example.artist.ui.topalbumslisting;

import com.example.artist.models.Album;

import java.util.List;


public interface TopAlbumsView {
    void showProgress();

    void hideProgress();

    void showError();

    void updateData(List<Album> topAlbums);

    void showEmpty();

    void hidEmpty();
}

File: Artist/app/src/main/java/com/example/artist/ui/topalbumslisting/TopAlbumsView.kt
package com.example.artist.ui.topalbumslisting

import com.example.artist.models.Album

interface TopAlbumsView {
    fun showProgress()
    fun hideProgress()
    fun showError()
    fun updateData(topAlbums: MutableList<Album>)
    fun showEmpty()
    fun hidEmpty()
}